### PR TITLE
Use `react-router-dom` Link for internal download link

### PR DESCRIPTION
**Pull request by mcbattirola**
The `SupportLink` used to display the downloads link was using a `a` tag with `target=_blank`, which makes the link open in a new tab.

Now, since the download link will route to an in-app page on Cloud, this case should use a Link, and open in the same tab.

This PR makes it behave as expected now, routing internally to the downloads page on cloud, and opening in a new tab for external links (non-cloud enterprise or OSS).

I've also renamed `SupportLink` to `ExternalSupportLink` to better reflect what it does and not sound like it is a unstyled version of `StyledSupportLink` (see https://github.com/gravitational/teleport/pull/37080#discussion_r1463607013)

`e` PR: https://github.com/gravitational/teleport.e/pull/3232

### CODE REVIEW

File: web/packages/teleport/src/Support/Support.tsx [hunk 1]
```diff
@@ -194,16 +192,33 @@ const getDocUrls = (version = '', isEnterprise: boolean) => {
...
+  isCloud,
+  isEnterprise,
+}: {
+  isCloud: boolean;
+  isEnterprise: boolean;
+}) => {
   if (isCloud) {
-    return cfg.routes.downloadCenter;
+    return (
+      <StyledSupportLink as={Link} to={cfg.routes.downloadCenter}>
```
Comment (zmb3) [inline web/packages/teleport/src/Support/Support.tsx:204]: Why do we use `StyledSupportLink` for cloud and `SupportLink` for enterprise?

Comment (mcbattirola) [reply web/packages/teleport/src/Support/Support.tsx:204]: `SupportLink` is just a small wrap around `StyledSupportLink`:

```
const SupportLink = ({ title = '', url = '' }) => (
  <StyledSupportLink href={url} target="_blank">
    {title}
  </StyledSupportLink>
);
```

Comment (zmb3) [reply web/packages/teleport/src/Support/Support.tsx:204]: So `SupportLink` is in fact styled. I find this a bit confusing.

Intuitively, if you have a `Foo` and a `StyledFoo`, I'm going to assume that `Foo` is equivalent to `UnstyledFoo`..

Comment (mcbattirola) [reply web/packages/teleport/src/Support/Support.tsx:204]: Yeah, that makes sense to me.

I think `SupportLink` is kind of unnecessary, each link can set its target. 

Or maybe we rename it to `ExternalSupportLink`, so it isn't as confusing between `SupportLink` and `StyledSupportLink`. Thoughts?

Comment (zmb3) [reply web/packages/teleport/src/Support/Support.tsx:204]: Either of those sound fine to me. No preference.

Comment (mcbattirola) [reply web/packages/teleport/src/Support/Support.tsx:204]: I went with the ExternalSupportLink route, thanks!

**Review by kimlisa — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**